nix/shells.nix

```diff
@@ -58,6 +58,7 @@ rec {
       gh
       nixpkgs-fmt
       cargo-sweep
+      go
```

Comment:
@steveeJ - is this the correct place to put a new build dependency for holochain?

Reply:
this just provides it to the developer shell.

native toolchains that are used during the build process go [here](https://github.com/holochain/holochain-nixpkgs/blob/341dc3d5ef1c45ba57af7493279858859a1d4d3a/packages/holochain/default.nix#L126)

we're in the process of switching our nix derivations for which i can't blindly suggest the place. which crate(s) will need the go binary available for its builds?

Reply:
@steveeJ - kitsune_p2p is the direct dependent, but everything downstream of that, including holochain will obviously need it too, then.

Reply:
hnrg... @steveeJ maybe it makes the most sense to wait for your refactor to finish before trying to get go into holonix?